processor/src/main/java/com/instagram/common/json/annotation/processor/JsonAnnotationProcessor.java

```diff
@@ -166,6 +168,27 @@ private void processClassAnnotation(Element element) {
     boolean abstractClass = false;
     TypeElement typeElement = (TypeElement) element;
 
+    // The annotation should be validated for an interface, but no code should be generated.
+    JsonType annotation = element.getAnnotation(JsonType.class);
+    if (element.getKind() == INTERFACE) {
+      if (StringUtil.isNullOrEmpty(annotation.serializeCodeFormatter())) {
```

Comment:
there is an option generateSerializer in JsonType to turn on and off the serializer. we should probably respect it here.

Reply:
Just to be clear: the current implementation works by inlining `serializeCodeFormatter` at generated call sites, the same as `valueExtractFormatter`. So to respect `generateSerializer` would mean adding some code to generate a serialization method here that we never actually use ourselves.

Reply:
ah.. good call. 

Not generating the serializer is probably a minor issue, I'm wondering how would we support read-only interface. It's going to be strange to ask them for serializeCodeFormatter(if they never expect the interface to be serialized). 

what's your thoughts? maybe we should move the null check(error message) to the later point when we actually need serialize code formatter?

Reply:
That idea sounds perfect. I'll make the change,

Reply:
I just realized that we can do the same thing for `valueExtractFormatter`. I don't see a reason why not.